### PR TITLE
New dark sky readings, loader, rake task, download as csv

### PR DESCRIPTION
**Pull request by jamesjefferies**

### CODE REVIEW

File: spec/system/school_onboarding_spec.rb [hunk 1]
```diff
@@ -44,6 +45,9 @@
       expect(page).to have_select('Weather Underground Data Feed Area', selected: 'BANES weather')
       expect(page).to have_select('Solar PV from The University of Sheffield Data Feed Area', selected: 'BANES solar')
 
+      # Why does this fail?!
+      # expect(page).to have_select('Dark Sky Data Feed Area', selected: 'BANES dark sky weather')
```
Comment (jamesjefferies) [inline spec/system/school_onboarding_spec.rb]: This is doing my nut! Any ideas @jamesalmond - I can't see the wood for the trees!

File: lib/tasks/data_feeds/dark_sky_temperature_loader.rake [hunk 2]
```diff
@@ -0,0 +1,37 @@
+namespace :data_feeds do
+  def process_area(area, start_date, end_date)
```
Comment (jamesalmond) [inline lib/tasks/data_feeds/dark_sky_temperature_loader.rake]: Possibly a bit much going on here for a rake task? It falls outside the bounds of our usual testing areas.

File: lib/tasks/data_feeds/dark_sky_temperature_loader.rake [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    )
+
+    # Data is returned as an array [[distance_to_weather_station, temperature_data, percent_bad, bad_data]
+    temperature_data = data[1]
+
+    temperature_data.each do |reading_date, temperature_celsius_x48|
+      next if temperature_celsius_x48.size != 48
+      record = DataFeeds::DarkSkyTemperatureReading.find_by(reading_date: reading_date, area_id: area.id)
+      if record
+        record.update(temperature_celsius_x48: temperature_celsius_x48)
```
Comment (jamesalmond) [inline lib/tasks/data_feeds/dark_sky_temperature_loader.rake]: Could we use an upsert here?

Comment (jamesjefferies) [reply lib/tasks/data_feeds/dark_sky_temperature_loader.rake]: We could - I wanted to look at the Rails 6 upsert functionality at some point.

File: lib/tasks/data_feeds/dark_sky_temperature_loader.rake (hunk 2 above)
Comment (jamesjefferies) [reply lib/tasks/data_feeds/dark_sky_temperature_loader.rake]: Yep, fair enough

File: lib/data_feeds/dark_sky_temperature_loader.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    def process_area(area)
+      data = @dark_sky_api_interface.historic_temperatures(
+        area.latitude,
+        area.longitude,
+        @start_date,
+        @end_date
+      )
+
+      # Data is returned as an array [[distance_to_weather_station, temperature_data, percent_bad, bad_data]
```
Comment (jamesalmond) [inline lib/data_feeds/dark_sky_temperature_loader.rb]: `[[` in this comment is confusing!

**Pull request merged**